crates/network/src/transaction/mod.rs

```diff
@@ -1,22 +1,21 @@
-use crate::Receipt;
 use alloy_primitives::{Bytes, ChainId, Signature, B256, U256};
-use alloy_rlp::{BufMut, Decodable, Encodable};
+use alloy_rlp::{BufMut, Encodable};
+
+mod common;
+pub use common::TxKind;
 
 mod signed;
 pub use signed::Signed;
 
 /// Represents a minimal EVM transaction.
-pub trait Transaction: Encodable + Decodable + Send + Sync + 'static {
+pub trait Transaction: std::any::Any + Encodable + Send + Sync + 'static {
```

Comment:
bleh do we need to explicitly declare the Any here? I guess it'll be always there for most types we impl transaction for, would add a note in the code on why it's needed